form-flow/src/test/kotlin/com/ritense/formflow/domain/FormFlowInstanceTest.kt

```diff
@@ -17,31 +17,86 @@
 package com.ritense.formflow.domain
 
 import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
+import com.ritense.formflow.domain.definition.FormFlowDefinition
+import com.ritense.formflow.domain.definition.FormFlowNextStep
+import com.ritense.formflow.domain.definition.FormFlowStep
+import com.ritense.formflow.domain.definition.FormFlowStepId
 import com.ritense.formflow.domain.instance.FormFlowInstance
 import com.ritense.formflow.domain.instance.FormFlowStepInstanceId
 import org.hamcrest.CoreMatchers.equalTo
 import org.hamcrest.MatcherAssert.assertThat
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.Assertions.assertNull
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.assertThrows
 
 internal class FormFlowInstanceTest {
 
     @Test
-    fun `complete should return new step id` () {
+    fun `complete should return new step` () {
+        val definition: FormFlowDefinition = mock()
+        val steps: Set<FormFlowStep> = mutableSetOf(
+            FormFlowStep(
+                FormFlowStepId.create("test"),
+                mutableListOf(FormFlowNextStep("123", "test2"))),
+            FormFlowStep(
+                FormFlowStepId.create("test2")
+            )
+        )
+
+        whenever(definition.startStep).thenReturn("test")
+        whenever(definition.steps).thenReturn(steps)
+
         val instance = FormFlowInstance(
-            formFlowDefinition = mock()
+            formFlowDefinition = definition
         )
 
         //complete task
-        instance.complete(instance.currentFormFlowStepInstanceId!!, "data")
+        val result = instance.complete(instance.currentFormFlowStepInstanceId!!, "data")
 
         assertThat(instance.getHistory()[0].submissionData, equalTo("data"))
+        assertNotNull(result)
+        assertEquals(result!!.stepKey, "test2")
```

Comment:
The expectation should be the first parameter. So it should be assertEquals("test2", result!!.stepKey)

Reply:
Whoops, I'll fix it!